### ISSUE TITLE
Sitemap.xml incomplete

### ISSUE THREAD

**Issue by professor-k** [nopSolutions/nopCommerce#6792]
nopCommerce version: 
4.60.3

Steps to reproduce the problem: 
It looks like #6349 wasn't fixed completely. Now under conditions specified in that issue (2 languages, between 25000 and 50000 urls) sitemap.xml only contains first half of the links (what should have been sitemap-1.xml).

**Comment by professor-k** [comment]
Apparently, the problem is here: https://github.com/nopSolutions/nopCommerce/blob/d05a04a6267d263c0fc37d09c6439638ac3fea9d/src/Presentation/Nop.Web/Factories/SitemapModelFactory.cs#L519C38-L519C38

There should be check for number of items in sitemaps list instead, that will fix the issue.

It might be worth separate issue, but while I investigated around this one, I figured that separate chunks of sitemap are generated independently, which is both suboptimal performance-wise (heavy GenerateUrlsAsync is called in every sitemap chunk, even that only part of it is written to the disk), and can lead to silly corner-case issues if set of urls changed between parts of sitemap are generated (i.e. there were exactly 50001 urls in the sitemap on the first call to sitemap.xml, index with two links was generated, then something changed that removed one url, and we'll end up with sitemap consisting of just index pointing to two crashing nonexistent chunks).
It would make sense to generate all the files in one go, as this will ensure both integrity of the sitemap and also will give an option to check for 50 MB file size constraint (which is not enforced as of now).
